### PR TITLE
d_a_do work

### PR DESCRIPTION
**Pull request by Pheenoh**
Many matching and nonmatching for the dog actor

### CODE REVIEW

File: include/JSystem/J3DGraphAnimator/J3DModel.h [hunk 1]
```diff
@@ -84,6 +97,7 @@ class J3DModel {
 
     // is there a better way to handle inlines with same name as non-inlines?
     MtxP i_getAnmMtx(int p1) { return mMtxBuffer->getAnmMtx(p1); }
+    void i_setAnmMtx(int p1, Mtx mtx) { return mMtxBuffer->setAnmMtx(p1, mtx); }
```
Comment (jdflyer) [inline include/JSystem/J3DGraphAnimator/J3DModel.h:100]: Do we need the return here?

File: include/rel/d/a/d_a_do/d_a_do.h [hunk 2]
```diff
@@ -2,5 +2,201 @@
...
+    /* 0x0C */ f32 mWalkSpeed;
+    /* 0x10 */ f32 mRunSpeed;
+    /* 0x14 */ f32 mSwimSpeed;
+    /* 0x18 */ f32 mPlayerRecogniztionDist;
+    /* 0x1C */ u8 field_0x1c;
+    /* 0x1D */ u8 mSwimming;
+    /* 0x1E */ u16 mWaterHuntAnimType;
+};
+
+struct some_class {
```
Comment (jdflyer) [inline include/rel/d/a/d_a_do/d_a_do.h:72]: Is this needed or just leftover from testing?

File: include/rel/d/a/d_a_do/d_a_do.h [hunk 3]
```diff
@@ -2,5 +2,201 @@
...
+    ACT_A_SWIM = 10,
+    ACT_SWIM = 20,
+    ACT_HELP,
+    ACT_BOAT,
+    ACT_HANG = 30,
+    ACT_CARRY = 35,
+    ACT_MESSAGE = 39
+};
+
+struct daDo_HIO_tmp {
```
Comment (TakaRikka) [inline include/rel/d/a/d_a_do/d_a_do.h:58]: is the temp needed instead of the actual class?

Comment (Pheenoh) [reply include/rel/d/a/d_a_do/d_a_do.h:58]: Using the actual class caused a __sinit function to not be generated? I couldn't figure it out

image

**Review by TakaRikka — APPROVED** [review]

**Review by jdflyer — APPROVED** [review]

**Pull request merged**